<launch>

  <include file="$(find jsk_perception)/sample/include/play_rosbag_apc2016_object_imgs_and_masks.xml" />

  <node name="feature_based_object_recognition"
        pkg="jsk_perception" type="feature_based_object_recognition.py"
        output="screen">
    <remap from="~input" to="publish_apc2016_object_imgs_and_masks/output/image" />
    <remap from="~input/mask" to="publish_apc2016_object_imgs_and_masks/output/mask" />
    <rosparam subst_value="true">
      pretrained_model: $(find jsk_perception)/trained_data/resnet152_from_caffe.npz
      mean_file: $(find jsk_perception)/trained_data/resnet_lsvrc2012_mean.npy
      db_file: $(find jsk_perception)/trained_data/resnet_features_apc2016.npz
    </rosparam>
  </node>

  <node name="draw_classification_result"
        pkg="jsk_perception" type="draw_classification_result.py">
    <remap from="~input/image" to="publish_apc2016_object_imgs_and_masks/output/image" />
    <remap from="~input" to="feature_based_object_recognition/output" />
    <rosparam>
      approximate_sync: true
    </rosparam>
  </node>
  <node name="tile_images"
        pkg="jsk_perception" type="tile_image.py">
    <rosparam>
      no_sync: true
      input_topics:
        - publish_apc2016_object_imgs_and_masks/output/image
        - publish_apc2016_object_imgs_and_masks/output/mask
        - draw_classification_result/output
    </rosparam>
  </node>
  <node name="image_view"
        pkg="image_view" type="image_view">
    <remap from="image" to="tile_images/output" />
  </node>

</launch>
